### DISCUSSION TITLE
a missing license for http.g4 grammar

### DISCUSSION THREAD

**Discussion by miryung** [antlr/grammars-v4#4233] [General]
This file does not have a license information. In the case of a missing license, is there a default license we can used for .g4 files under antler/grammars-v4?

 https://github.com/antlr/grammars-v4/tree/master/http

**Comment by kaby76** [comment]
No license generally means that you can't use it, modify it, or do anything with it. But, this contradicts the purpose, the reason why this repo exists.

The proper way to proceed is to ask the original author for a license. It may be impossible to find the author as well. The issue was raised before. https://github.com/antlr/grammars-v4/issues?q=is%3Aissue++license+ What should have been done was to force the author to choose a license for all new grammars. Otherwise, the PR should be summarily rejected. It's not worth any of our hard work to do anything at all with an unlicensed work.

**Comment by Marti2203** [comment]
Hi, I am the creator of http.g4 and I just opened a PR #4237  with an MIT license. 
Thanks @miryung for contacting me! :)
The file was created 5 years ago when the control for some these things was lower.
